spec/requests/api/cycles_spec.rb

```diff
@@ -38,4 +38,37 @@
       end
     end
   end
+
+  path '/api/cycles/{id}' do
+    get 'Returns Cycle details' do
+      tags 'Cycles'
+      description 'Retrieves the details of a cycle'
+      parameter name: :id, in: :path, type: :string
+      produces 'application/json'
+      operationId 'getCycle'
+
+      let(:curriculum_id) { create(:curriculum).id }
+      let(:cycle) { create(:cycle, curriculum_id:) }
+      let(:id) { cycle.id }
+
+      response '200', 'Success' do
+        schema type: :object,
+               properties: {
+                 id: { type: :integer },
+                 name: { type: :string },
+                 order_number: { type: :integer },
+                 learning_goals_description: { type: :string },
+                 challenge_description: { type: :string },
+                 boilerplate_url: { type: :string }
+               }
+
+        run_test!
+      end
+
+      response '404', 'Cycle not found' do
```

Comment:
I'm not sure if you should test Unauthorized response. It depends if this access is public or not.

Reply:
you won't need to be logged in to access this